core/src/main/java/tech/tablesaw/analytic/AggregateFunctions.java

```diff
@@ -213,41 +233,188 @@ public Double getValue() {
+
+        @Override
+        public void addRightMost(T newValue) {
+          if (Double.isNaN(sum)) {
+            this.sum = 0.0;
+          }
+          this.sum += newValue.doubleValue();
+          count++;
+        }
+      };
     }
 
     @Override
-    AggregateFunction<T, Double> slidingFunction() {
-      throw new UnsupportedOperationException("Analytic Function Mean Is Not implemented");
+    AggregateFunction<T, Double> functionForSlidingWindows() {
+      return new AggregateFunction<T, Double>() {
+        private final ArrayDeque<Double> queue = new ArrayDeque<>();
+        private Double sum = 0.0;
+        private int missingCount = 0;
+
+        @Override
+        public void removeLeftMost() {
+          Double removed = queue.remove();
+          if (Double.isNaN(removed)) {
+            missingCount--;
+          } else {
+            this.sum -= removed;
+          }
+        }
+
+        @Override
+        public void addRightMost(T newValue) {
+          Double doubleValue = newValue.doubleValue();
+          this.sum += doubleValue;
+          queue.add(doubleValue);
+        }
+
+        @Override
+        public void addRightMostMissing() {
+          queue.add(Double.NaN);
+          missingCount++;
+        }
+
+        @Override
+        public Double getValue() {
+          if (queue.size() - missingCount == 0) {
+            return Double.NaN;
+          }
+          return sum / (queue.size() - missingCount);
+        }
+      };
     }
   }
 
-  static class Count<T extends Number> extends WindowDependentAggregateFunction<T> {
+  static class Count<T> extends WindowDependentAggregateFunction<T> {
 
     @Override
-    AggregateFunction<T, Double> fixedFunction() {
-      throw new UnsupportedOperationException("Analytic Function Count Is Not implemented");
+    AppendAggregateFunction<T, Integer> functionForAppendWindows() {
+      return new AppendAggregateFunction<T, Integer>() {
+        private int count = 0;
+
+        @Override
+        public Integer getValue() {
+          return count;
+        }
+
+        @Override
+        public void addRightMostMissing() {}
+
+        @Override
+        public void addRightMost(T newValue) {
+          count++;
+        }
+      };
     }
 
     @Override
-    AggregateFunction<T, Double> slidingFunction() {
-      throw new UnsupportedOperationException("Analytic Function Count Is Not implemented");
+    AggregateFunction<T, Integer> functionForSlidingWindows() {
+      return new AggregateFunction<T, Integer>() {
+        // True if is missing value;
+        private final ArrayDeque<Boolean> queue = new ArrayDeque<>();
+        private int missingCount = 0;
+
+        @Override
+        public void removeLeftMost() {
+          Boolean removedMissingValue = queue.remove();
+          if (removedMissingValue) {
+            missingCount--;
+          }
+        }
+
+        @Override
+        public void addRightMost(T newValue) {
+          queue.add(false);
```

Comment:
i guess i'm starting to understand. :) but this is a public method, so how do we know that newValue is not missing?

Reply:
This method has a public modifier because it overrides the interface, but the interface is package private and a user will never see these implementation classes or methods.